OctaneVSPlugin/ConnectionSettings.cs

```diff
@@ -0,0 +1,246 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using MicroFocus.Adm.Octane.Api.Core.Connector;
+using MicroFocus.Adm.Octane.Api.Core.Connector.Authentication;
+using MicroFocus.Adm.Octane.VisualStudio.Common;
+using MicroFocus.Adm.Octane.VisualStudio.View;
+using MicroFocus.Adm.Octane.VisualStudio.ViewModel;
+using Microsoft.VisualStudio.Shell;
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using static System.Threading.Tasks.Task;
+
+namespace MicroFocus.Adm.Octane.VisualStudio
+{
+	/// <summary>
+	/// Options page represent the settings the user can set for ALM Octane plugin.
+	/// This class is presented to the user as a page in Visual Studio options dialog.
+	/// </summary>
+	[Guid("1D9ECCF3-5D2F-4112-9B25-264596873DC9")]
+	internal class ConnectionSettings : UIElementDialogPage
+	{
+
+		public const string ConnectionSuccessful = "Connection successful!";
+		private string url = string.Empty;
+		private int ssid = 1001;
+		private int wsid = 1002;
+		private string user = string.Empty;
+		private string password = string.Empty;
+		private bool credentialLogin = true;
+		private bool ssologin = false;
+		private ConnectionSettingsView page;
+
+		protected override void OnApply(PageApplyEventArgs e)
+		{
+
+			page.SetInfoLabelText("");
+			SaveSettingsToStorage();
+			var result = Run(async () => { return await TestConnection(); }).Result;
+
+			if (!result.Equals(ConnectionSuccessful))
+			{
+				e.ApplyBehavior = ApplyKind.CancelNoNavigate;
+			}
+
+			else
+			{
+				// reset and thus require a new octane service obj
+				Run(async () => { return await OctaneServices.Reset(); }).Wait();
+
+				// create a new service object
+				OctaneServices.Create(OctaneConfiguration.Url,
+						  OctaneConfiguration.SharedSpaceId,
+						  OctaneConfiguration.WorkSpaceId);
+
+                
+                // close all opened details windows so that we don't have details windows
+                // for entities from different workspaces
+                PluginWindowManager.CloseAllDetailsWindows();
+
+				// disable the active item toolbar because we don't know yet
+				// whether we can connect with the new Octane credentials
+				MainWindowCommand.Instance.DisableActiveItemToolbar();
+
+				// After settings are applied we notify the main ViewModel to allow it to refresh.
+				if (OctaneMyItemsViewModel.Instance != null)
+				{
+                    InitialisePluginComponents();
+				}
+
+			}
+
+			page.SetInfoLabelText(result);
+			base.OnApply(e);
+		}
+
+        public async void InitialisePluginComponents()
+        {
+            await OctaneServices.GetInstance().Connect();
+            await OctaneMyItemsViewModel.Instance.LoadMyItemsAsync();
+            await EntityTypeRegistry.Init();
+        } 
+
+		public async Task<string> TestConnection()
```

Comment:
indentation driving me crazy

Reply:
the file is messed up for some reason by github